### ISSUE TITLE
Wildcards support for sub-file-paths and audio-file-paths

### ISSUE THREAD

**Issue by snoopy-coder** [mpc-qt/mpc-qt#132]
Is it possible to add _wildcards_ support for the `sub-file-paths` and `audio-file-paths` parameters, as is done, for example, in the [fuzzydir.lua](https://github.com/sibwaf/mpv-scripts?tab=readme-ov-file#fuzzydirlua) script? Thanks a lot. 

![mpc-qt-sub](https://github.com/user-attachments/assets/7307f756-c156-4584-9d03-8977719ad2b5)

![mpc-qt-audio](https://github.com/user-attachments/assets/c82cda07-5f8a-4167-8727-8973fd01527e)

**Comment by cmdrkotori** [comment]
Yeah, it's possible.  I guess if I were to intervene in how mpv parses those provided paths, the code would use `glob()` and `globfree()`, or the equivalent in Qt should I keep to the Qt libs.

About the script though: I can't embed that script because it uses PowerShell, which isn't usually available on Linux PCs.  Besides, there really is no need for such a script when mpv could evaluate such paths with a small patch; functions for evaluating globs is natively available in C/C++ code, no lua code needed.

**Comment by snoopy-coder** [comment]
Thank you for the quick and detailed response!
Let me clarify right away that I'm not asking you to add support for LUA or PowerShell scripts. I mentioned `fuzzydir.lua` only to describe the desired functionality.  
I would just like to have a proper search for external audio tracks and subtitles, without having to rename and copy to certain folders every time, otherwise the player does not find them. 
I guess this can be implemented using a couple of additional options in the sections in the screenshots above:

1. A **checkbox** with _"Search also in subdirectories"_ 
2. A **spinbox** with a numeric value of the _maximum search depth_ (usually it is not too large).

And then process it in any way that you consider the most convenient and cross-platform.

PS. Is it probably better to change the title by removing the mention of wildcards from there?
